### ISSUE TITLE
Paging doesn't recalculate total results when filtering

### ISSUE THREAD

**Issue by tdreyno** [holidayextras/jsonapi-server#171]
I am using the MemoryHandler currently, so let me know if that's the source of my issue, but:

I'm running a simple filter on a collection.

```
/comments?filter[authorName]=Thomas+Reynolds
```

The response correctly returns the only item where this filter is true. Unfortunately, the `meta` paging information still shows the original count:

```
"meta":{"page":{"offset":0,"limit":50,"total":8000}
```

Shouldn't that say `1` rather than `8000` (the original collection length) in this filtering context? Or am I misunderstanding the spec.

**Comment by theninj4** [comment]
Hi @tdreyno - You're correct the memoryHandler doesn't update the pagination count. The reason why it doesn't work is because the memoryHandler doesn't actually do any filtering on resources, it lets jsonapi-server's post middleware handle it. It does, however, report back the total number of resources it *thinks* match (all of them) which leads to the odd behaviour you're seeing.

The other memory handler's I've implemented all handle filtering which means the numbers all match up. The test suit (which is reused in the handler modules) actually looks for discrepancies between the number of results returned vs the number of claimed resources in the pagination block, which is a great indicator that a handler isn't filtering properly.

I could remove the opt-in-to-pagination within the memoryHandler, however that'd mean anyone trying the memory handler via the built-in example server won't see the pagination block and might not even be aware it supports pagination.

In a nutshell, pagination is the only thing in the memoryHandler that doesn't function correctly from an end-user perspective,  sorry.

**Comment by tdreyno** [comment]
@theninj4 I also just implemented by own memory handler to do that.

That said, having the default example on the project have such a serious bug might be worth resolving. Either through documentation or by removing the broken functionality in favor of saying "implementers should handle this in their handler"

**Comment by theninj4** [comment]
I see where you're coming from, but I don't see it as a serious bug - I can't think of any good reasons to use the memory handler in a customer-facing production environment.

You're right that the documentation for handlers should cover pagination, I'll get that resolved :+1:

**Comment by tdreyno** [comment]
No, but it's perfectly reasonable to use the memory handler in a prototype or testing scenario, both of which would benefit from pagination actually working.

But it's cool. Your project, I'll let it be.

**Comment by adamlc** [comment]
@theninj4 I'm developing my own handler and have this issue. Is there a way I should be handling this properly without having to check all the filter stuff myself in my own handler?

**Comment by theninj4** [comment]
I don't think there's any way to correctly handle pagination if your handler doesn't support it. It'd be possible to correctly identify the size of the page, but all other functionality will be broken.

You might be better off opting out of pagination entirely by dropping the `resultCount` from your search function. Theoretically you should be able to simply drop that param, although a quick test suggests there might be a bug there - if you pass it as `null` it'll behave as expected.

**Comment by adamlc** [comment]
Cool thanks! I have an external service that returns very fuzzy results, so quite a few of them are useless and later removed by the filters. I guess not having pagination isn't an issue for this particular endpoint at least!
